lib/stdlib/src/supervisor.erl

```diff
@@ -257,7 +265,7 @@ call(Supervisor, Req) ->
       ChildSpecs :: [child_spec()],
       Result :: 'ok' | {'error', Error :: term()}.
 check_childspecs(ChildSpecs) when is_list(ChildSpecs) ->
-    case check_startspec(ChildSpecs) of
+    case check_startspec(ChildSpecs, undefined) of
```

Comment:
Maybe add a `check_childspecs/2` so the `auto_shutdown` value can be given.

Reply:
Sure, why not :) I would keep the `undefined` option, though, and make it the default for `check_childspecs/1` instead of defaulting to the `auto_shutdown` default, `never`, so the function would keep the semantics of "are those childspecs valid?" instead of changing it to "are those childspecs valid _for the default auto_shutdown_?"

Reply:
Added `check_childspecs/2` in last commit